src/App.tsx

```diff
@@ -77,7 +65,9 @@ export const App: React.FC = () => {
                   <PostsList
                     posts={posts}
                     selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
+                    onPostSelected={post =>
+                      dispatch(postsActions.setSelectedPost(post))
```

Comment:
Consider using a named function for the `onPostSelected` handler instead of an inline arrow function. This can improve readability and make it easier to test or reuse the function.